libcnb/src/layer_env.rs

```diff
@@ -482,7 +483,7 @@ impl LayerEnvDelta {
         self.entries
             .get(&(ModificationBehavior::Delimiter, key.into()))
             .cloned()
-            .unwrap_or_default()
+            .unwrap_or(OsString::from(PATH_LIST_SEPARATOR))
```

Comment:
Problem - This code is used in two ways: Writing env var changes to CNB layers and reading env var changes to CNB layers. 

We can make assumptions or change defaults about the writing because we "own" the interface and expectations there. However, we cannot change expectations about reading as we don't know who wrote it. We could work through/around this decoupling this logic somehow either on write or read. 

For example, on read if the delim doesn't exist, deserialize that as an explicit `""` (empty string) until the upstream spec is updated, and never write a layer without a `.delim` (by defaulting to `;`).